CorpusCallosum/cc_visualization.py

```diff
@@ -0,0 +1,152 @@
+        - "red_to_yellow": Red -> Yellow -> Light Blue -> Blue
+        - "yellow_to_red": Yellow -> Light Blue -> Blue -> Red
+
+        color_range: Optional tuple of (min, max) to set fixed color range for the colorbar
+        twoD: If True, generate 2D visualization instead of 3D mesh
+    """
+    # Convert paths to Path objects
+    contours_path = Path(contours_path) if contours_path is not None else None
+    thickness_path = Path(thickness_path)
+    measurement_points_path = Path(measurement_points_path)
+    output_dir = Path(output_dir)
+
+    # Load data and create mesh
+    cc_mesh = CC_Mesh(num_slices=1)  # Will be resized when loading data
+
+    if contours_path is not None:
+        cc_mesh.load_contours(str(contours_path))
+    else:
+        cc_contour, anterior_endpoint_idx, posterior_endpoint_idx = load_fsaverage_cc_template()
+        cc_mesh.contours[0] = np.stack(cc_contour).T
+        cc_mesh.start_end_idx[0] = [anterior_endpoint_idx, posterior_endpoint_idx]
+
+    cc_mesh.load_thickness_values(str(thickness_path), str(measurement_points_path))
+    cc_mesh.set_resolution(resolution)
+
+    if twoD:
+        # cc_mesh.smooth_contour(contour_idx=0, window_size=5)
+        cc_mesh.plot_cc_contour_with_levelsets(
+            contour_idx=0, levelpaths=None, title=None, save_path=str(output_dir / "cc_thickness_2d.png"), colorbar=True
+        )
+    else:
+        cc_mesh.fill_thickness_values()
+        # Create and process mesh
+        cc_mesh.create_mesh(smooth=smooth_iterations, closed=False)
+
+        # Generate visualizations
+        cc_mesh.plot_mesh(
+            colormap=colormap,
+            color_range=color_range,
+            thickness_overlay=True,
+            show_contours=False,
+            show_mesh_edges=True,
+            legend=legend,
+        )
+        cc_mesh.plot_mesh(str(output_dir / "cc_mesh.html"), thickness_overlay=True)
+
+        cc_mesh.plot_cc_contour_with_levelsets(
+            contour_idx=len(cc_mesh.contours) // 2, save_path=str(output_dir / "midslice_2d.png")
+        )
+
+        cc_mesh.to_fs_coordinates(vox_size=[resolution, resolution, resolution])
+        cc_mesh.write_vtk(str(output_dir / "cc_mesh.vtk"))
+        cc_mesh.write_fssurf(str(output_dir / "cc_mesh.fssurf"))
+        cc_mesh.write_overlay(str(output_dir / "cc_mesh_overlay.curv"))
+        cc_mesh.snap_cc_picture(str(output_dir / "cc_mesh_snap.png"))
+
+
+if __name__ == "__main__":
+    options = options_parse()
+    main_args = {
+        "contours_path": options.contours,
+        "thickness_path": options.thickness,
+        "measurement_points_path": options.measurement_points,
+        "output_dir": options.output_dir,
+        "resolution": options.resolution,
+        "smooth_iterations": options.smooth_iterations,
+        "colormap": options.colormap,
+        "color_range": options.color_range,
+        "legend": options.legend,
+        "twoD": options.twoD,
+    }
```

Comment:
```suggestion
    options = make_parser().parse_args()
    sys.exit(main(
        contours_path=options.contours,
        thickness_path=options.thickness,
        measurement_points_path=options.measurement_points,
        output_dir=options.output_dir,
        resolution=options.resolution,
        smooth_iterations=options.smooth_iterations,
        colormap=options.colormap,
        color_range=ptions.color_range,
        legend=options.legend,
        twoD=options.twoD,
    ))
```

To make this work, main should return `0` or the error message.